src/main/java/io/stargate/sgv2/jsonapi/service/shredding/model/AtomicValue.java

```diff
@@ -25,6 +26,11 @@ public static AtomicValue forNumber(BigDecimal num) {
     return create(DocValueType.NUMBER, num.toPlainString());
   }
 
+  public static AtomicValue forTimestamp(Date dt) {
+    // Long-value of timestamp still fits without truncation
+    return create(DocValueType.TIMESTAMP, String.valueOf(dt.getTime()));
```

Comment:
Not sure if matters but why are we storing the long as a String here?

Reply:
This is for Digest value ("hash"); we cannot have polymorphism, so everything ends up as String, when stored in `array_contains`: which is used for equality filtering (except for special case of document id). So same done for every value type during shredding (Arrays and Objects are composed of digests of leaf values along with path).